lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3InputStream.java

```diff
@@ -123,6 +123,12 @@ public long skip(long n)
         ensureOpen();
         seekStream();
 
+        if ((length != null) && (streamPosition + n > length)) {
```

Comment:
I've stumbled upon this while running `io.trino.filesystem.s3.TestS3FileSystemS3Mock#testInputFile()` test